manualsource/src/main/java/org/hibernate/search/manualsource/FullTextQuery.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.search.manualsource;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.lucene.search.Explanation;
```

Comment:
Can we avoid Lucene exposure in the API? It'll be a general theme in HS 6, but it'd be good to avoid it in newly written code to begin with.